distributed/tests/test_client.py

```diff
@@ -6741,3 +6741,21 @@ def f(x):
 
     c_default = default_client()
     assert c is c_default
+
+
+@gen_cluster(client=True)
+async def test_annotate_collections(c, s, a, b):
+    import dask.array as da
+
+    x = da.ones(10)
+
+    with dask.annotate(foo="bar"):
+        y = x + 1
+
+    assert y.__dask_graph__().layers[y.name].annotations == {"foo": "bar"}
+
+    y = await y.persist()
```

Comment:
I suspect this is from https://github.com/dask/dask/issues/7036 (note that adding `optimize_graph=False` will result in this test passing).